airflow/providers/amazon/aws/hooks/redshift_statement.py

```diff
@@ -0,0 +1,144 @@
+
+    :param redshift_conn_id: reference to
+        :ref:`Amazon Redshift connection id<howto/connection:redshift>`
+    :type redshift_conn_id: str
+
+    .. note::
+        get_sqlalchemy_engine() and get_uri() depend on sqlalchemy-amazon-redshift
+    """
+
+    conn_name_attr = 'redshift_conn_id'
+    default_conn_name = 'redshift_default'
+    conn_type = 'redshift'
+    hook_name = 'Amazon Redshift'
+    supports_autocommit = True
+
+    @staticmethod
+    def get_ui_field_behavior() -> Dict:
+        """Returns custom field behavior"""
+        return {
+            "hidden_fields": [],
+            "relabeling": {'login': 'User', 'schema': 'Database'},
+        }
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    def _get_conn_params(self) -> Tuple[Dict[str, Union[str, int]], Dict]:
+        """Helper method to retrieve connection args and kwargs"""
+        conn = self.get_connection(
+            self.redshift_conn_id  # type: ignore[attr-defined]  # pylint: disable=no-member
+        )
+
+        conn_params: Dict[str, Union[str, int]] = {
+            "user": conn.login or '',
+            "password": conn.password or '',
+            "host": conn.host or '',
+            "port": conn.port or 5439,
+            "database": conn.extra_dejson.pop('database', ''),
+        }
+
+        conn_kwargs: Dict = {}
+
+        for param, arg in conn.extra_dejson.items():
+            conn_kwargs[param] = arg
+
+        return conn_params, conn_kwargs
+
+    def get_uri(self) -> str:
+        """
+        Override DbApiHook get_uri method for get_sqlalchemy_engine()
+
+        .. note::
+            Value passed to connection extra parameter will be excluded
+            from returned uri but passed to get_sqlalchemy_engine()
+            by default
+        """
+        from sqlalchemy.engine.url import URL
+
+        conn_params, _ = self._get_conn_params()
+        conn_params['username'] = conn_params['user']
+        del conn_params['user']
+
+        return URL(drivername='redshift+redshift_connector', **conn_params).__to_string__(hide_password=False)
+
+    def get_sqlalchemy_engine(self, engine_kwargs=None):
+        """Overrides DbApiHook get_sqlalchemy_engine to pass redshift_connector specific kwargs"""
+        _, conn_kwargs = self._get_conn_params()
+        if engine_kwargs is None:
+            engine_kwargs = {}
+
+        if "connect_args" in engine_kwargs:
+            engine_kwargs["connect_args"] = {**conn_kwargs, **engine_kwargs["connect_args"]}
+        else:
+            engine_kwargs["connect_args"] = conn_kwargs
+
+        return super().get_sqlalchemy_engine(engine_kwargs=engine_kwargs)
+
+    def get_conn(self) -> RedshiftConnection:
+        """Returns a redshift_connector.Connection object"""
+        conn_params, conn_kwargs = self._get_conn_params()
+        conn_kwargs: Dict = {**conn_params, **conn_kwargs}
+        self.log.info(conn_kwargs)
```

Comment:
Is this log necessary? It looks like it might log the connection password from `conn_params`

Reply:
And if you _do_ want a log, this should probably be debug-level and have a more identifiable format, e.g.

```python
self.log.debug("Redshift connection arguments: %s", conn_kwargs)
```

Reply:
addressed in 42d38c4